PR: https://github.com/odoo/odoo/pull/

From: e27340715785187467e3b3385fb4682e580122b7
From: Martin Geubelle
Date: 2018-09-03 11:40:22

Structural Changes: 4
Total Changes: 18

[FIX] mail: consider `activity` as a valid view type

The recently introduced `activity` view (see rev. odoo/odoo@1de9496 has not
received all the consideration it deserves as a real view type (no RNG, not in
the valid view `type`, no doc, etc.). It was currently only working based on a
`_get_default_activity_view` method.

Even if the RNG is currently really basic, its grammar could evolve in the future.

This was causing an issue when opening Studio with this view as a default view
with an invalid type was created.

================================= pseudo patch: =================================

--- a/addons/mail/models/__init__.py
+++ b/addons/mail/models/__init__.py
@@ -19,6 +19,8 @@ from . import res_users
 from . import res_company
 from . import res_config_settings
 from . import update
+from . import ir_action_act_window
 from . import ir_actions
 from . import ir_autovacuum
 from . import ir_model
+from . import ir_ui_view

--- a/None
+++ b/addons/mail/models/ir_action_act_window.py
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+from odoo import fields, models
+
+
+class ActWindowView(models.Model):
+    _inherit = 'ir.actions.act_window.view'
+
+    view_mode = fields.Selection(selection_add=[('activity', 'Activity')])

--- a/None
+++ b/addons/mail/models/ir_ui_view.py
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+from odoo import fields, models
+
+
+class View(models.Model):
+    _inherit = 'ir.ui.view'
+
+    type = fields.Selection(selection_add=[('activity', 'Activity')])
